src/hooks/useContract.ts

```diff
@@ -132,8 +138,13 @@ export function useSocksController(): Contract | null {
   )
 }
 
-export function useHALORewardsContract(): Contract | null {
+export function useHALORewardsContract(rewardsVersion = 1): Contract | null {
```

Comment:
In typescript, shouldnt we do something like

```suggestion
export function useHALORewardsContract(rewardsVersion : number = 1): Contract | null {
```

Reply:
The type is already "inferred" by the assigned value so its not required to specify the type.

I personally prefer this because there is less code to write/read, makes the code cleaner & easier to understand.